samples/snippets/src/main/java/com/example/bigquerystorage/WriteBufferedStream.java

```diff
@@ -0,0 +1,100 @@
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class WriteBufferedStream {
+
+  public static void runWriteBufferedStream()
+      throws DescriptorValidationException, InterruptedException, IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "MY_PROJECT_ID";
+    String datasetName = "MY_DATASET_NAME";
+    String tableName = "MY_TABLE_NAME";
+
+    writeBufferedStream(projectId, datasetName, tableName);
+  }
+
+  public static void writeBufferedStream(String projectId, String datasetName, String tableName)
+      throws DescriptorValidationException, InterruptedException, IOException {
+    try (BigQueryWriteClient client = BigQueryWriteClient.create()) {
+      // Initialize a write stream for the specified table.
+      // For more information on WriteStream.Type, see:
+      // https://googleapis.dev/java/google-cloud-bigquerystorage/latest/com/google/cloud/bigquery/storage/v1beta2/WriteStream.Type.html
+      WriteStream stream = WriteStream.newBuilder().setType(WriteStream.Type.BUFFERED).build();
+      TableName parentTable = TableName.of(projectId, datasetName, tableName);
+      CreateWriteStreamRequest createWriteStreamRequest =
+          CreateWriteStreamRequest.newBuilder()
+              .setParent(parentTable.toString())
+              .setWriteStream(stream)
+              .build();
+      WriteStream writeStream = client.createWriteStream(createWriteStreamRequest);
+
+      // Use the JSON stream writer to send records in JSON format.
+      // For more information about JsonStreamWriter, see:
+      // https://googleapis.dev/java/google-cloud-bigquerystorage/latest/com/google/cloud/bigquery/storage/v1beta2/JsonStreamWriter.html
+      try (JsonStreamWriter writer =
+          JsonStreamWriter.newBuilder(writeStream.getName(), writeStream.getTableSchema())
+              .build()) {
+        // Write two batches to the stream, each with 10 JSON records.
+        for (int i = 0; i < 2; i++) {
+          JSONArray jsonArr = new JSONArray();
+          for (int j = 0; j < 10; j++) {
+            // Create a JSON object that is compatible with the table schema.
+            JSONObject record = new JSONObject();
+            record.put("col1", String.format("buffered-record %03d", i));
+            jsonArr.put(record);
+          }
+          ApiFuture<AppendRowsResponse> future = writer.append(jsonArr);
+          AppendRowsResponse response = future.get();
+        }
+
+        // Flush the buffer.
+        FlushRowsRequest flushRowsRequest =
+            FlushRowsRequest.newBuilder()
+                .setWriteStream(writeStream.getName())
+                .setOffset(Int64Value.of(10 * 2 - 1)) // Advance the cursor to the latest record.
+                .build();
+        FlushRowsResponse flushRowsResponse = client.flushRows(flushRowsRequest);
```

Comment:
Maybe add another write without flush and comment that those rows will not appear in the table.

Reply:
IMO this would some unneeded complexity to the sample, and also make it harder to compare the differences with pending and committed modes. Also it would make the code not strictly correct, if a user didn't read the code comments carefully.